modules/agent/pkg/deployer/manager.go

```diff
@@ -62,7 +74,7 @@ func (m *Manager) Cleanup() error {
 		bundleDeployment, err := m.bundleDeploymentCache.Get(m.fleetNamespace, deployed.BundleID)
 		if apierror.IsNotFound(err) {
 			logrus.Infof("Deleting orphan bundle ID %s, release %s", deployed.BundleID, deployed.ReleaseName)
-			if err := m.deployer.Delete(deployed.BundleID, ""); err != nil {
+			if err := m.deployer.Delete(deployed.BundleID, deployed.ReleaseName); err != nil {
```

Comment:
This is the place where bundle is deleted and we need to clean up release. If this is empty then we will try to delete release with the bundleId, which could leave orphan releases since bundleID is not the same as releaseName.

Reply:
@ibuildthecloud